treewide: cleanups preparing release

PolynomialDivision opened the pull request.

PolynomialDivision committed:
ath25: 5:15: copy config and patch from 5.10

Copy config and patch from kernel 5.10 to kernel 5.15.

Signed-off-by: Nick Hainke <vincent@systemli.org>

PolynomialDivision committed:
ath25: 5.15: enable 5.15 testing kernel

Manually refreshed:
- 140-redboot_boardconfig.patch
- 141-redboot_partition_scan.patch
- 142-redboot_various_erase_size_fix.patch

Automatically refreshed:
- 107-ar5312_gpio.patch
- 108-ar2315_gpio.patch
- 110-ar2313_ethernet.patch
- 120-spiflash.patch
- 130-watchdog.patch
- 330-board_leds.patch

Use "make kernel_oldconfig" to refresh the new kernel config.

The Ubiquiti Nanostation 2 (XS2) and Ubiquiti Nanostation 5 (XS5) should
be marked as broken when switching to 5.15 by default. The new kernel
does not fit anymore into the partition.

Signed-off-by: Nick Hainke <vincent@systemli.org>

alexeys85 committed:
omap: switch default kernel to 5.15

Switch to kernel 5.15 although it is marked as source only. It is marked
as source only because the eth0 interface can't be added to the br-lan
default interface resulting in a broken ethernet [0]. There is a
workaround to set a higher pvd (e.g. to 4000) in the dts [1].

Other functionality was already tested on a Beaglebone Black for
+6month now without issues.

Merging this is important to finally branch a new release. We want to
get rid completely of 5.10 kernel. Without this commit we get following
error: "Missing kernel version/hash file for 5.10." Hopfully someone
will fix the board in the rc phase.

- [0] https://github.com/openwrt/openwrt/issues/11953
- [1] https://github.com/openwrt/openwrt/issues/11953#issuecomment-1419078990

Signed-off-by: Alexey Smirnov <s.alexey@gmail.com>
[rewrite commit message]
Signed-off-by: Nick Hainke <vincent@systemli.org>

PolynomialDivision committed:
treewide: remove files for building 5.10 kernel

All targets are bumped to 5.15. Remove the old 5.10 patches, configs
and files using:

  find target/linux -iname '*-5.10' -exec rm -r {} \;

Further, remove the 5.10 include.

Signed-off-by: Nick Hainke <vincent@systemli.org>

PolynomialDivision committed:
toolchain: glibc: increase min kernel version to 5.15

The purpose of this is to remove all of the compatibility code that is
normally compiled into glibc for kernels < 5.15 from our build. Only
Linux kernels >= 5.15 will be supported.

Signed-off-by: Nick Hainke <vincent@systemli.org>

PolynomialDivision committed:
kernel: remove unnecessary qca-sdk patch for 5.10 kernel

We removed 5.10 kernel, so remove also the patch that only affects 5.10
kernels.

Manually refresh:
- 0005-SSDK-config-add-kernel-5.15.patch
- 0010-QSDK-config-Avoid-Werror-heroics.patch

Signed-off-by: Nick Hainke <vincent@systemli.org>

PolynomialDivision committed:
treewide: remove all @{lt,gt,ge,le} appearances affecting 5.10 kernel

We only use 5.15 kernel. So remove all those unnecessary appearances.

Signed-off-by: Nick Hainke <vincent@systemli.org>

john-tho commented:
A few extras still around:
```
find target/linux -iname '*-5.10'
target/linux/ipq806x/files-5.10
target/linux/bcm4908/files-5.10
target/linux/realtek/dts-5.10
target/linux/realtek/files-5.10
target/linux/generic/pending-5.10
target/linux/generic/hack-5.10
target/linux/generic/backport-5.10
```

The make `DEPENDS`ed and config vars `LINUX_5_10` & `LINUX_5_15`
`git grep -E 'LINUX_5_1(0|5)'`

https://github.com/openwrt/openwrt/pull/12531/commits/b54c6617f881239918f1ef201644921c832fc7fc
>Remove the old 5.10 patches and configs except ath25, lantiq and omap

The commit message is not correct? Those paths got removed as well.

PolynomialDivision committed:
treewide: cleanup kernel symbol references

We only use 5.15 kernel. So remove all those unnecessary symbols
referencing 5.10 or 5.15 kernel.

Can be found with:
  git grep -E 'LINUX_5_1(0|5)'

Note that we remove the dependency from "sound-soc-chipdip-dac" instead
of removing the complete kernel package. The 5.15 version bump forgot to
delete the "@LINUX_5_10" dependency. The kernel package is still needed
in 5.15 kernel.

Signed-off-by: Nick Hainke <vincent@systemli.org>

john-tho commented:
There are a few [in-tree drivers that use `KERNEL_VERSION`](https://github.com/search?q=repo%3Aopenwrt%2Fopenwrt+LINUX_VERSION_CODE+path%3A%2F%5Etarget%5C%2Flinux%5C%2F%2F&type=code) (and an awful lot of patches), but it might be more work than it is worth to adjust them. I guess we could come up with a https://coccinelle.lip6.fr/ semantic patch for the C files. Worth waiting to hear some more opinions before starting on that.

guidosarducci commented:
> > Also, it might be premature to drop the LINUX_5_15 references, and instead review them as part of kernel 6.1 integration work.
> 
> Could you please explain this in more detail? I am not sure what the advantages are of keeping the references?
> 
@PolynomialDivision I was thinking changes like https://github.com/openwrt/openwrt/pull/12531/commits/c1f1e145e3a59639b23dc669ffa4ccd87ef00cad#diff-53028509d59d66b72d97f2686b4f03f8c79fd2d5cabd1701d75cd5aa9f9a4483 and https://github.com/openwrt/openwrt/pull/12531/commits/ceb7d5f200408a7c95dac8f162912833a25af143#diff-53028509d59d66b72d97f2686b4f03f8c79fd2d5cabd1701d75cd5aa9f9a4483 affect both 5.15 and 6.1 builds. If you're checking the upstream Kconfigs and Makefiles yourself to ensure things will build properly on both 5.15 and 6.1 then that's all fine. If not, then perhaps better to leave these updates to 6.1 merging/build testing, especially considering overlap with @Ansuel 's 6.1 branch (e.g. 188cbc370d415312f88e88b08bd9cf3f31b1482e).

DragonBluep commented on target/linux/ramips/image/mt7621.mk:
```diff
@@ -9,10 +9,6 @@ DEFAULT_SOC := mt7621
 DEVICE_VARS += ELECOM_HWNAME LINKSYS_HWNAME DLINK_HWID
 
-ifdef CONFIG_LINUX_5_10
-  DTS_CPPFLAGS += -DDTS_LEGACY
```
This flag is used in "mt7621.dtsi", please clean it as well.

PolynomialDivision replied:
thanks!

PolynomialDivision committed:
ramips: remove device tree legacy compatibility

We switched to 5.15 kernel, so we don't need to maintain 5.10
compatibility anymore.

Signed-off-by: Nick Hainke <vincent@systemli.org>

1715173329 commented on package/kernel/qca-ssdk/patches/0001-SSDK-config-add-kernel-5.10.patch:
```diff
@@ -1,56 +0,0 @@
-From 6ee395e1cab32d1d4899d7846ea99e6e2dc0a9e4 Mon Sep 17 00:00:00 2001
```
Need to refresh`0005-SSDK-config-add-kernel-5.15.patch` as it is built after this patch.

PolynomialDivision replied:
Indeed. Thanks!

PolynomialDivision committed:
ath25: switch default kernel to 5.15

Switch to kernel 5.15 although it is marked as source only. It is marked
as source only because nobody was able to test this device.

Merging this is important to finally branch a new release. We want to
get rid completely of 5.10 kernel. Without this commit we get following
error: "Missing kernel version/hash file for 5.10." Hopfully someone
will fix the board in the rc phase or it will be removed completly.

Signed-off-by: Nick Hainke <vincent@systemli.org>

PolynomialDivision commented:
> If you're checking the upstream Kconfigs and Makefiles yourself to ensure things will build properly on both 5.15 and 6.1 then that's all fine.

I just checked everything, if the symbols and names are still the same. However, I am not sure how to find the kmod dependencies of the different kmods. Is there a good way to check that?

PolynomialDivision commented:
I would say this is ready.

Ansuel commented:
@PolynomialDivision nice cleanup... I wonder if we should make a list of thing to check when these kind of thing has to happen. What do you think? And maybe put it in the wiki?

PolynomialDivision commented:
> What do you think? And maybe put it in the wiki?

Yes of course. Documentation is always important. :)

abajk commented:
It is still possible to remove some unused code:
~~~
$ grep -rni "LINUX_VERSION_CODE" target/ | grep -v ">= KERNEL_VERSION(5,15,0)"
...
target/linux/generic/files/drivers/net/phy/ar8216.c:894:#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0)
target/linux/generic/files/drivers/net/phy/ar8216.c:906:#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0)
target/linux/generic/files/drivers/net/phy/b53/b53_common.c:509:#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0)
target/linux/generic/files/drivers/net/phy/b53/b53_common.c:515:#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 5, 0)
target/linux/generic/files/block/partitions/fit.c:76:#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0)
target/linux/ramips/files/drivers/pinctrl/pinctrl-aw9523.c:389:#if LINUX_VERSION_CODE <= KERNEL_VERSION(5, 5, 0)
target/linux/ramips/files/drivers/pinctrl/pinctrl-aw9523.c:440:#if LINUX_VERSION_CODE <= KERNEL_VERSION(5, 5, 0)
target/linux/ramips/files/drivers/net/ethernet/ralink/mdio.c:67:#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
target/linux/ramips/files/drivers/net/ethernet/ralink/mdio.c:80:#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
target/linux/ramips/files/drivers/net/ethernet/ralink/soc_mt7620.c:147:#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
target/linux/ramips/files/drivers/net/ethernet/ralink/soc_mt7620.c:181:#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
target/linux/ramips/files/drivers/net/ethernet/ralink/mdio_rt2880.c:159:#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
target/linux/ramips/files/drivers/net/ethernet/ralink/mdio_rt2880.c:179:#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)
target/linux/ramips/files/drivers/net/ethernet/ralink/mtk_eth_soc.c:1099:#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 0)

~~~

neheb commented:
should probably add the ksmbd change here.

PolynomialDivision commented:
> should probably add the ksmbd change here.

Which change?

PolynomialDivision commented:
> It is still possible to remove some unused code:

@john-tho already suggsted that. :)

neheb commented:
> > should probably add the ksmbd change here.
> 
> Which change?

https://github.com/openwrt/openwrt/pull/10296

Just keep the 5.15 changes.

Ansuel commented:
@PolynomialDivision ok to merge this? any problem that may arise or it's safe?

PolynomialDivision commented:
> ok to merge this?

I would say yes :)

> any problem that may arise or it's safe?

At least I don't see any issues.
